src/main/java/seedu/trackascholar/model/major/MajorContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,43 @@
+package seedu.trackascholar.model.major;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.trackascholar.commons.util.StringUtil;
+import seedu.trackascholar.model.applicant.Applicant;
+
+/**
+ * Tests if any of the {@code Applicant}'s {@code Major} matches any of the keywords given.
+ */
+public class MajorContainsKeywordsPredicate implements Predicate<Applicant> {
+    private final List<String> keywords;
+
+    public MajorContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Applicant applicant) {
+        Set<String[]> majors = applicant.getMajors()
```

Comment:
Is there a reason why the type is `Set<String[]>` instead of `Set<String>`? Seems like converting the set to a list would be easier if the type was `Set<String>` instead.